TextReplacementHandler.apxc

```diff
@@ -0,0 +1,1771 @@
+/** TextReplacementHandler
+** Created by Krizia on 04/10/2018
+** 
+** Purpose of Class: 
+** Take an input string and replace merge fields with actual values 
+** Can be used with the Outbound Email module to send merged Email Templates 
+** 
+** Prerequisites / Items to Note when using: 
+** Must update the constructor to query the object(s) you need 
+** Set consistent delimiters (e.g. "{!" as opening and "}" as closing) 
+** Run using batch class if needing to run many times (due to queries) 
+** 
+** Steps to Implement in Setup 
+** n/a
+**
+How to call: ** The Id you send in should be what links all objects/records together for the merge fields to work. (in example: the Account Id) ** 
+    TextReplacementHandler trh = new TextReplacementHandler(theId); 
+    String replaceMe = 'This is what needs to get replaced with the Account Name: {!Account.Name} ' + 
+        'and this is what needs to get replaced with the Owner name: {!User.Name} and again here we ' + 
+        'will have the Account Name: {!Account.Name}. '; 
+    String replaced = ''; 
+    replaced = trh.populateMergeFields(replaceMe, '{!', '}'); 
+    System.debug(replaceMe); 
+    System.debug(replaced); 
+**
+**/
+public class TextReplacementHandler {
+
+
+
+    //**** All necessary objects here **** 
+
+    public sObject obj1 { get; set; }
+    public Account theAccount { get; set; } 
+    public User theUser { get; set; }
+
+    //**** All necessary objects here **** 
+
+
+    
+    public TextReplacementHandler(Id recordId) {
+        //**** Get all necessary objects here **** 
+        try {
+     //     Example:   
+            sObject s1 = getAllFields(recordId, 'Account'); 
+            theAccount = new Account(); 
+            theAccount = (Account)s1;
+            System.debug(theAccount); 
+    
+            sObject s2 = getAllFields( theAccount.OwnerId, 'User'); 
+            theUser = new User(); 
+            theUser = (User)s2; 
+            System.debug(theUser); 
+     
+
+        }            
+        //**** Get all necessary objects here **** 
+        catch (Exception e) {
+            System.debug(e);
+            system.debug(e.getMessage());
+            system.debug(e.getLineNumber());
+            system.debug(e.getStackTraceString());
+        }
+    } //public TextReplacementHandler() 
+
+    
+
+    /** Query all fields from a record - can be used separately from Text Replacement 
+    ** Input: the Id of the record you need and what object it is (string) 
+    ** Output: sObject of the object type and with the record Id you chose with all of its fields (cast the returned record) 
+    **/
+    public static sObject getAllFields(Id recordId, String obj) { 
```

Comment:
@KriziaRC this method name kinda confused me - I thought it was here to just "get the list of fields for an object", but it's also "querying out the data for the object"... suggest breaking it up and renaming...